### PR TITLE
Feature/batch create labels and page labels

### PR DESCRIPTION
**Pull request by kakj-go**
#### What type of this PR
/kind feature

#### What this PR does / why we need it:
batch create labels and page labels

### CODE REVIEW

File: modules/pipeline/dbclient/op_pipeline_labels.go [hunk 1]
```diff
@@ -63,6 +71,27 @@ func (client *Client) CreatePipelineLabels(p *spec.Pipeline, ops ...SessionOptio
...
 }
 
+func (client *Client) PagePipelineLabels(req *apistructs.PipelineLabelListRequest, ops ...SessionOption) ([]spec.PipelineLabel, int64, error) {
+	session := client.NewSession(ops...)
+	defer session.Close()
+
+	var labels []spec.PipelineLabel
+	sql := session.Table(spec.PipelineLabel{}.TableName())
+	if len(req.PipelineSource) > 0 {
+		sql = sql.Where("pipeline_source = ?", req.PipelineSource)
```
Comment (sfwn) [inline modules/pipeline/dbclient/op_pipeline_labels.go:81]: pipeline source must specified.

File: modules/pipeline/dbclient/op_pipeline_labels.go [hunk 2]
```diff
@@ -63,6 +71,27 @@ func (client *Client) CreatePipelineLabels(p *spec.Pipeline, ops ...SessionOptio
...
+	}
+	if len(req.KeyNotLike) > 0 {
+		sql = sql.Where("`key` not like ?", "%"+req.KeyNotLike+"%")
```
Comment (sfwn) [inline modules/pipeline/dbclient/op_pipeline_labels.go:84]: Use `key` IN (xxx, yyy) to match index.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/endpoints/pipeline_label.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		return errorresp.ErrResp(fmt.Errorf("auth error: not internal client"))
+	}
+
+	if len(createReq.Labels) <= 0 {
+		return apierrors.ErrCreatePipelineLabel.InvalidParameter("labels").ToResp(), nil
+	}
+
+	for _, label := range createReq.Labels {
+		if label.TargetID <= 0 {
+			return apierrors.ErrCreatePipelineLabel.InvalidParameter("label targetID").ToResp(), nil
```
Comment (sfwn) [inline modules/pipeline/endpoints/pipeline_label.go:55]: label [i] missing targetID

File: apistructs/pipeline_label.go [hunk 4]
```diff
@@ -52,3 +56,35 @@ type TargetIDSelectByLabelRequest struct {
 	// OrderByTargetIDAsc 根据 target_id 升序，默认为 false，即降序
 	OrderByTargetIDAsc bool `json:"orderByTargetIDAsc"`
 }
+
+type PipelineLabelBatchInsertResponse struct{}
```
Comment (sfwn) [inline apistructs/pipeline_label.go:60]: useless

File: modules/pipeline/endpoints/pipeline_label.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	identityInfo, err := user.GetIdentityInfo(r)
+	if err != nil {
+		return errorresp.ErrResp(err)
+	}
+	if !identityInfo.IsInternalClient() {
+		return errorresp.ErrResp(fmt.Errorf("auth error: not internal client"))
+	}
+
+	if len(req.PipelineYmlName) <= 0 {
+		return apierrors.ErrListPipelineLabel.InvalidParameter("pipelineYmlName").ToResp(), nil
```
Comment (sfwn) [inline modules/pipeline/endpoints/pipeline_label.go:97]: missing xxx

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/pipeline_label.go [hunk 6]
```diff
@@ -52,3 +56,35 @@ type TargetIDSelectByLabelRequest struct {
...
+
+type PipelineLabelBatchInsertRequest struct {
+	Labels []PipelineLabel `json:"labels"`
+}
+
+type PipelineLabelListRequest struct {
+	PipelineSource  PipelineSource `json:"pipelineSource"`
+	PipelineYmlName string         `json:"pipelineYmlName"`
+	TargetIDS       []uint64       `json:"targetIds"`
```
Comment (sfwn) [inline apistructs/pipeline_label.go:69]: TargetIDs is required or not?

File: apistructs/pipeline_label.go [hunk 7]
```diff
@@ -52,3 +56,35 @@ type TargetIDSelectByLabelRequest struct {
...
+	Keys            []string       `json:"keys"`
+
+	FilterTargetIDNotEqualKeys []string `json:"filterTargetIDNotEqualKeys"`
```
Comment (sfwn) [inline apistructs/pipeline_label.go:72]: split into two request.
one request with `Keys` field for dbclient.
another request with `FilterTargetIDNotEqualKeys` for api use.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/pipeline_label.go (hunk 7 above)
Comment (sfwn) [reply apistructs/pipeline_label.go:72]: MatchKeys
UnMatchKeys

File: apistructs/pipeline_label.go [hunk 8]
```diff
@@ -52,3 +56,32 @@ type TargetIDSelectByLabelRequest struct {
...
 }
+
+type PipelineLabelBatchInsertRequest struct {
+	Labels []PipelineLabel `json:"labels"`
+}
+
+type PipelineLabelListRequest struct {
+	PipelineSource  PipelineSource `schema:"pipelineSource" json:"pipelineSource"`
+	PipelineYmlName string         `schema:"pipelineYmlName" json:"pipelineYmlName"`
+	TargetIDs       []uint64       `schema:"targetIds" json:"targetIds"`
```
Comment (sfwn) [inline apistructs/pipeline_label.go:67]: In tag you should use schema:"targetID", and invoker will use this query params like: /api/xxx?targetID=1&targetID=2

use ID, not Id.

File: apistructs/pipeline_label.go [hunk 9]
```diff
@@ -52,3 +56,32 @@ type TargetIDSelectByLabelRequest struct {
...
+	MatchKeys       []string       `schema:"matchKeys" json:"matchKeys"`
```
Comment (sfwn) [inline apistructs/pipeline_label.go:68]: Same issue.
Use `matchKey` in schema and this library will collect all values with same key in query params to `MatchKeys` slice.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/pipeline_label.go (hunk 8 above)
Comment (sfwn) [reply apistructs/pipeline_label.go:67]:
```golang
TargetIDs       []uint64       `schema:"targetID"`
```

No need json tag.

fdp client: /api/pipeline-labels?targetID=1&targetID=2&targetID=3

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/pipeline_label.go (hunk 8 above)
Comment (kakj-go) [reply apistructs/pipeline_label.go:67]: https://stackoverflow.com/questions/11944410/passing-array-in-get-for-a-rest-call
use targetIds not a bad idea, user see end with s then he will know he can send a array params

Comment (kakj-go) [reply apistructs/pipeline_label.go:67]: add json tag does not affect use

**Review by sfwn — APPROVED** [review]

**Pull request merged**